exporter/exporter.go

```diff
@@ -25,119 +21,29 @@ type cat1data struct {
-
-		var mappedDc = mappedDataCriteria[dc]
-		if mappedDc.FieldOids == nil {
-			mappedDc = mdc{DataCriteria: dataCriteria, FieldOids: make(map[string][]string)}
-		}
-
-		// Add all the codedValues onto the list of field OIDs
-		for field, descr := range dataCriteria.FieldValues {
-			if descr.Type == "CD" {
-				mappedDc.FieldOids[field] = append(mappedDc.FieldOids[field], descr.CodeListID)
-			}
-		}
-
-		// If the data criteria has a negation, add the reason onto the returned FieldOids
-		if dataCriteria.Negation {
-			mappedDc.FieldOids["REASON"] = append(mappedDc.FieldOids["REASON"], dataCriteria.NegationCodeListID)
-		}
-
-		// If the data criteria has a value, and it's a "coded" type, added the CodeListId into the result OID set
-		if dataCriteria.Value.Type == "CD" {
-			mappedDc.ResultOids = append(mappedDc.ResultOids, dataCriteria.CodeListID)
-		}
-
-		if dc.DataCriteriaOid != "" {
-			mappedDataCriteria[dc] = mappedDc
-		}
-	}
-
-	// Add the key to the value to get what HDS would have returned
-	var retDataCriteria []mdc
-	for key, value := range mappedDataCriteria {
-		value.DataCriteriaOid = key.DataCriteriaOid
-		value.ValueSetOid = key.ValueSetOid
-		retDataCriteria = append(retDataCriteria, value)
-	}
-	return retDataCriteria
-}
-
-func exporterFuncMap(cat1Template *template.Template) template.FuncMap {
+func exporterFuncMap(cat1Template *template.Template, vsMap models.ValueSetMap) template.FuncMap {
 	return template.FuncMap{
-		"timeNow":                      time.Now().UTC().Unix,
-		"newRandom":                    uuid.NewRandom,
-		"timeToFormat":                 timeToFormat,
-		"identifierForInt":             identifierForInt,
-		"identifierForString":          identifierForString,
-		"escape":                       escape,
-		"entryInfosForPatient":         entryInfosForPatient,
-		"executeTemplateForEntry":      generateExecuteTemplateForEntry(cat1Template),
-		"condAssign":                   condAssign,
-		"valueOrNullFlavor":            valueOrNullFlavor,
-		"dischargeDispositionDisplay":  dischargeDispositionDisplay,
-		"sdtcValueSetAttribute":        sdtcValueSetAttribute,
-		"getTransferOid":               getTransferOid,
-		"identifierForInterface":       identifierForInterface,
-		"codeToDisplay":                codeToDisplay,
-		"valueOrDefault":               valueOrDefault,
-		"reasonValueSetOid":            reasonValueSetOid,
-		"oidForCodeSystem":             oidForCodeSystem,
-		"oidForCode":                   oidForCode,
+		"timeNow":             time.Now().UTC().Unix,
+		"newRandom":           uuid.NewRandom,
+		"timeToFormat":        timeToFormat,
+		"identifierForInt":    identifierForInt,
+		"identifierForString": identifierForString,
+		"escape":              escape,
+		// mattrbianchi: I searched for this and it wasn't used in any of the templates... should it really be in here?
+		//"entryInfosForPatient":         models.Record.EntryInfosForPatient,
```

Comment:
You can get rid of this, if it's not used in the templates. We'll re-add it if we need it.